sqlglot/dialects/clickhouse.py

```diff
@@ -578,6 +578,28 @@ def _parse_index_constraint(
                 granularity=granularity,
             )
 
+        def _parse_partition(self) -> t.Optional[exp.Partition]:
+            # https://clickhouse.com/docs/en/sql-reference/statements/alter/partition#how-to-set-partition-expression
+            if not self._match(TokenType.PARTITION):
```

Comment:
It looks like both `PART` and `PARTITION` are valid keywords, so we can do the following here:

```python
if not self._match_texts(("PART", "PARTITION")):
    return None
```

Reply:
I feel like that the `PART` should be separated from `PARTITION`. The current implementation does store the used keyword, so generating SQL will yield `PARTITION` every time. Moreover, not every `ALTER` partition statement supports working with parts. And unfortunately, the doc is poorly written and the `PART` expression supports only string literals:

```sql
alter table tmp.foo drop part tuple(toYYYYMM(toDate('2019-01-25')))
```

```
DB::Exception: Syntax error: failed at position 31 ('tuple'): tuple(toYYYYMM(toDate('2019-01-25'))). Expected string literal. (SYNTAX_ERROR)
```

```sql
alter table tmp.foo drop part id '1'
```

```
DB::Exception: Syntax error: failed at position 31 ('id'): id '1'. Expected string literal. (SYNTAX_ERROR)
```

Reply:
Oh they have different semantics? I thought they simply allowed it as an abbreviation, is that not the case?

Reply:
Yeah, the part is basically part of data on the disk, and partition is logical subset of parts with the same partition key

Reply:
Gotcha, thanks for clarifying. Looks good then.